apps/legacy-api/testing/LegacyStubServer.ts

```diff
@@ -21,10 +23,36 @@ export class LegacyStubServer extends RootServer {
     const adapter = new FastifyAdapter({
       logger: false
     })
-    return module.createNestApplication<NestFastifyApplication>(adapter)
+    const app = module.createNestApplication<NestFastifyApplication>(adapter)
+    this.recordAllRegisteredRoutes(app)
+    return app
   }
 
   async init (app: NestFastifyApplication, config: ConfigService): Promise<void> {
     await app.init()
   }
+
+  /**
+   * Helper to get all the registered routes for testing purposes
+   */
+  getAllRoutes (): RegisteredRoute[] {
+    return this.allRoutes
+  }
+
+  private recordAllRegisteredRoutes (app: NestFastifyApplication): void {
+    app.getHttpAdapter()
+      .getInstance()
+      .addHook('onRoute', (opts: RegisteredRoute) => {
+        this.allRoutes.push(opts)
+      })
+  }
+}
+
+// See https://www.fastify.io/docs/latest/Reference/Hooks/#onroute
```

Comment:
```suggestion
/**
 * @see https://www.fastify.io/docs/latest/Reference/Hooks/#onroute
 */
```

should stick with JSDoc format when possible